cli/pcluster/config/validators.py

```diff
@@ -1484,3 +1499,27 @@ def ebs_volume_size_snapshot_validator(section_key, section_label, pcluster_conf
                     )
                 )
     return errors, warnings
+
+
+def validate_storage_type_options(
+    errors, fsx_storage_type, fsx_deployment_type, fsx_per_unit_storage_throughput, fsx_drive_cache_type
+):
+    if fsx_storage_type == "HDD":
+        if fsx_deployment_type != "PERSISTENT_1":
+            errors.append("For HDD filesystems, 'deployment_type' must be 'PERSISTENT_1'")
+        if fsx_per_unit_storage_throughput not in FSX_HDD_THROUGHPUT:
+            errors.append(
+                "For HDD filesystems, 'per_unit_storage_throughput' can only have the following values: {0}".format(
+                    FSX_HDD_THROUGHPUT
+                )
+            )
+    elif fsx_storage_type != "HDD" and fsx_drive_cache_type is not "NONE":
```

Comment:
I'd limit this `if/else` to the two cases HDD and SSD and avoid the `!= HDD` one.

In this way in the SSD case, if both the conditions are not satisfied, we can provide both the errors to the users:
* _'drive_cache_type' features can be used only with HDD filesystems_ and
* _For SSD filesystems, 'per_unit_storage_throughput' can only have the following values_

Instead by having a different `elif` branch we are showing only one of the two messages.

We can write:
```python
if fsx_storage_type == "HDD":
    ...
else:  # SSD
    if fsx_drive_cache_type is not "NONE":
       errors.append("'drive_cache_type' features can be used only with HDD filesystems")
    if fsx_per_unit_storage_throughput and fsx_per_unit_storage_throughput not in FSX_SSD_THROUGHPUT:
        errors.append(
            "For SSD filesystems, 'per_unit_storage_throughput' can only have the following values: {0}".format(
                FSX_SSD_THROUGHPUT
            )
        )
```